### PR TITLE
#184347034 implement wishing a product

### PR DESCRIPTION
**Pull request by Munezeroolivierhugue**
### What does this PR do?

Add product to wishlist

### Description of Task to be completed?

- Adding product to wishlist
- Removing product to wishlist
- Getting wishes made for a product
- Getting wishes made by certain user 

### How should this be manually tested?

1. Clone [repo](https://github.com/atlp-rwanda/vikings-ec-bn.git)
2. Change into the directory vikings-ec-bn
3. Checkout to the branch: feature/add-wishlist-184347034
4. Run the command `npm install` to install all packages
5. Run `npm run seeding`
6. Run `npm run dev`
7. Add products to wishlist at `http://localhost:<port>/api/v1/product-wishes`  URL in Postman

#### What are the relevant pivotal tracker stories?

[#184347034](https://www.pivotaltracker.com/story/show/184347034)

### CODE REVIEW

File: src/controllers/wishlist.controller.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { CartService } from "../services/cart.service";
+import { ProductService } from "../services/product.service";
+import { WishService } from "../services/wishlist.service";
+
+//adding wishes to wishlist
```
Comment (irakozetony) [inline src/controllers/wishlist.controller.js]: remove unnecessary comments

File: src/controllers/wishlist.controller.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+export class WishlistController {
+  static async addToWishlist(req, res) {
+    try {
+      const { id: userId } = req.user;
+      let product_id = req.body.productId;
+      let wish = req.wishlist;
+      if(!product_id){
```
Comment (irakozetony) [inline src/controllers/wishlist.controller.js]: this code should be in validation

File: src/controllers/wishlist.controller.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        message: "Item added to wishlist successfully",
+      });
+    } catch (error) {
+      return res.status(500).json({
+        error: error.message,
+        message: "Could not add product to wishlist, try again",
+      });
+    }
+  }
+  //getting wishes by product
```
Comment (irakozetony) [inline src/controllers/wishlist.controller.js]: remove unnecessary comment

File: src/middlewares/product.middleware.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ export const checkProductAvailable = async (req, res, next) => {
   });
   if (!getProduct)
     return res.status(404).json({ message: 'Product not found' });
+    req.product = getProduct;
```
Comment (Abdden) [inline src/middlewares/product.middleware.js:23]: Please, format your codes correctly.

File: src/services/wishlist.service.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { Wishlist } from "../database/models/index";
+import { ProductService } from "./product.service";
+export class WishService {
+    //get wish by user_id
```
Comment (irakozetony) [inline src/services/wishlist.service.js]: remove unnecessary comments

**Review by irakozetony — CHANGES_REQUESTED** [review]

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
Rename your commit to match the convetions

**Comment by irakozetony** [comment]
Change your commit message

**Review by irakozetony — APPROVED** [review]

**Review by nshimiyejayd — APPROVED** [review]

**Review by fabmukunzi — APPROVED** [review]

**Review by istafrica-dennw — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**